src/net_processing.cpp

```diff
@@ -3566,20 +3622,20 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         // to users' AddrMan and later request them by sending getaddr messages.
         // Making nodes which are behind NAT and can only make outgoing connections ignore
         // the getaddr message mitigates the attack.
-        if (!pfrom.IsInboundConn()) {
+        if (!RelayAddrsWithPeer(*peer)) {
```

Comment:
oh wow. That's terrible. Thank you for catching this.

This was a bad rebase on top of 49c10a9ca40, which combined the two error conditions:

```diff
diff --git a/src/net_processing.cpp b/src/net_processing.cpp
index 690b59476b..859b67755d 100644
--- a/src/net_processing.cpp
+++ b/src/net_processing.cpp
@@ -3521,11 +3521,7 @@ void PeerManager::ProcessMessage(CNode& pfrom, const std::string& msg_type, CDat
         // Making nodes which are behind NAT and can only make outgoing connections ignore
         // the getaddr message mitigates the attack.
         if (!pfrom.IsInboundConn()) {
-            LogPrint(BCLog::NET, "Ignoring \"getaddr\" from outbound connection. peer=%d\n", pfrom.GetId());
-            return;
-        }
-        if (!pfrom.RelayAddrsWithConn()) {
-            LogPrint(BCLog::NET, "Ignoring \"getaddr\" from block-relay-only connection. peer=%d\n", pfrom.GetId());
+            LogPrint(BCLog::NET, "Ignoring \"getaddr\" from %s connection. peer=%d\n", pfrom.ConnectionTypeAsString(), pfrom.GetId());
             return;
         }
```

Reply:
Oh, I see. There should be a functional test for this that would have failed - something which I [once](https://github.com/bitcoin/bitcoin/pull/19315#issuecomment-701571168) wanted to do anyway but then forgot about. I'll  write one soon!

Reply:
I'll very happily review once you open the PR.

Reply:
heads up that I've recently written some similar tests here: https://github.com/bitcoin/bitcoin/pull/21528/files?file-filters%5B%5D=.py#diff-5a9f30817894348260893f195cea07da88271b5e01b1531f0cd839ee85898592R150. I don't think it would cover this exact use case, but just mentioning incase its useful / incase the test improvements could play nice together.

Reply:
Cool - I didn't know, I had only looked at the non-test commits in your PR! This use case is the reverse direction (inbound/outbound send the getaddr to our node and get back addrs/nothing) from your tests, so I agree that these belong together.

Reply:
More tests added in #21707